[WPI] DB設計の見直し

yukri1030 opened the pull request:
# WHAT
chat, commentを無くし、messageというテーブルに変更。

has-througを無くしました。

# WHY
message同士のやり取りを行うアソシエーションをどのように組むべきか分かりません

yukri1030 committed:
Merge pull request #1 from yukri1030/DB_product

[WIP]　DBの設計図を考える

yukri1030 committed:
[WPI] DB設計の見直し

# WHAT
chat, commentを無くし、messageというテーブルに変更。

has-througを無くしました。

# WHY
message同士のやり取りを行うアソシエーションをどのように組むべきか分かりません

TOMARUMARU reviewed (commented):
アソシエーションがまだ不十分なので、今一度見直してください。
中間テーブルを用いた多対多のアソシエーションを使用します

yukri1030 committed:
[WPI] DBの設計図

# WHAT
messageテーブルの他に、それと多 対 多になる、commentテーブル、それらの中間テーブル（conversationテーブル）を作成しました。

また、group-nameやuser-nameには一意性制約のuniqu:trueを追加しました

以前と変わっていないのですが、memberテーブルのアソシエーションは、groupしか指定していません（member_id = user_id）だと思うのですが、言及していません

# WHY
上記に対するコメントと、他、間違っている点（細かいところなど）を指摘してくださると幸いです

TOMARUMARU commented:
多対多はメッセージとは関係ありません。
一度メンターに質問してDB設計についてアドバイスをもらいながら修正しましょう。

また、DB設計のPRなのでReadMe以外のファイルはコミットしないでください

yukri1030 commented:
すみません。ありがとうございます！メンターに聞いてみます。

あの、このDB設計が終わった後の方が、先のカリキュラムであるhamlなどを進める形になるのでしょうか。

yukri1030 committed:
[WPI]DB設計

# WHAT
中間テーブルとして、memberテーブルを指定。それを元に、アソシエーションを見直しました。

# WHY
正しいアソシエーションは組めているでしょうか。

yukri1030 committed:
[WPI]DB設計

# WHAT
中間テーブルとして、memberを設定。
それを元に、アソシエーションも見直しました。

#WHY
間違っている部分があれば、ご指摘願いあmす。

yukri1030 committed:
[WPI] chat画面の実装

# WHAT
chat-spaceのトップ画面を実装しました

# WHY
hamlは一つにまとめ、
sassは、画面の右側（_right.scss）と左側（index.sass）で分けました。

色や高さなどは、同じものでも、数回しか使っていないため、@mixinでまとめてはいません。

本当は、もう少し部分テンプレートを使って、すっきりとさせたいのですが、親要素の継承の仕方がわからず、レイアウトが崩れてしまうため、現在のような長いコードになっています。

アドバイスお願いします。

yukri1030 committed:
[WPI] chat-space フロントの実装

#  WHAT
何
chat-spaceのトップ画面を実装しました

＃#  なぜ
hamlはひとつまとめ、
sassは、画面の右側（_right.scss）と左側（index.sass）で分けました。

色や高さなどは、同じものでも、何回使っていないため、@mixinでまとめてはせません。

本当に、もう少し部分テンプレートを使って、ゆっくりとさせたいのですが、親要素の継承の仕方がわからず、レイアウトが崩れてしまった、現在のような長いコードになっています。

アドバイスをお願いします。
